website/docs/reference/global-configs.md

```diff
@@ -103,6 +103,41 @@ config:
 
 </File>
 
+<VersionBlock firstVersion="1.1">
+
+### Cache database objects for selected resource
+
+:::caution Experimental config flag
+This should be considered an _unstable_ config flag because it is experimental and subject to change. We reserve the right to make breaking changes to this config flag.
+:::
+
+When true, dbt caches database objects relevant to selected resource. Otherwise, dbt will cache all database objects related to the project.
```

Comment:
We're missing some more generic documentation about why caching exists and how it works, akin to the Notion doc I shared in a previous comment. In the meantime, it might be worth an extra bit of explanation here.
```suggestion
At the start of runs, dbt caches metadata about all the objects in all the schemas where it might materialize resources (such as models). By default, dbt caches all schemas related to the project. When this config is enabled, dbt will only cache schemas related to selected resources for the current run. This can offer significant speed improvements when running a small subset of a large project.
```